src/pikepdf/_methods.py

```diff
@@ -571,6 +573,99 @@ def _attach(self, *, basename, filebytes, mime=None, desc=''):  # pragma: no cov
+        release that object when appropriate.
+
+        Examples:
+
+            >>> with Pdf.open("test.pdf") as pdf:
+                    ...
+
+            >>> pdf = Pdf.open("test.pdf", password="rosebud")
+
+        Args:
+            filename_or_stream (os.PathLike): Filename of PDF to open
+            password (str or bytes): User or owner password to open an
+                encrypted PDF. If the type of this parameter is ``str``
+                it will be encoded as UTF-8. If the type is ``bytes`` it will
+                be saved verbatim. Passwords are always padded or
+                truncated to 32 bytes internally. Use ASCII passwords for
+                maximum compatibility.
+            hex_password (bool): If True, interpret the password as a
+                hex-encoded version of the exact encryption key to use, without
+                performing the normal key computation. Useful in forensics.
+            ignore_xref_streams (bool): If True, ignore cross-reference
+                streams. See qpdf documentation.
+            suppress_warnings (bool): If True (default), warnings are not
+                printed to stderr. Use :meth:`pikepdf.Pdf.get_warnings()` to
+                retrieve warnings.
+            attempt_recovery (bool): If True (default), attempt to recover
+                from PDF parsing errors.
+            inherit_page_attributes (bool): If True (default), push attributes
+                set on a group of pages to individual pages
+            access_mode (pikepdf.AccessMode): If ``.default``, pikepdf will
+                decide how to access the file. Currently, it will always
+                selected stream access. To attempt memory mapping and fallback
+                to stream if memory mapping failed, use ``.mmap``.  Use
+                ``.mmap_only`` to require memory mapping or fail
+                (this is expected to only be useful for testing). Applications
+                should be prepared to handle the SIGBUS signal on POSIX in
+                the event that the file is successfully mapped but later goes
+                away.
+            allow_overwriting_input (bool): If True, allows to edit the input
+                file by substituting it with a temporary BytesIO object under
+                the hood.
+            save_on_close (bool): Automatically calls ``.save()`` when
+                ``.close()`` is invoked. Requires ``allow_overwriting_input=True``.
+        Raises:
+            pikepdf.PasswordError: If the password failed to open the
+                file.
+            pikepdf.PdfError: If for other reasons we could not open
+                the file.
+            TypeError: If the type of ``filename_or_stream`` is not
+                usable.
+            FileNotFoundError: If the file was not found.
+        """
+        if allow_overwriting_input and not Path(filename_or_stream).is_file():
```

Comment:
I suspect `Path(filename_or_stream)` may throw an exception for certain streams, so we trap that.

Reply:
Code improved